docs/architecture.md

```diff
@@ -0,0 +1,292 @@
+
+```
+Armada selected as cluster manager.
+ArmadaClientApplication.start() called!
+host is localhost, port is 50051
+Submit health good!
+Executor JobID: 01jt6bmrszmnk4apxby6e895d8  Error: None
+Executor JobID: 01jt6bmrszmnk4apxby6xrwer6  Error: None
+Executor JobID: 01jt6bmrt0bvv5dh3k5k1qc8ta  Error: None
+Executor JobID: 01jt6bmrt0bvv5dh3k5mqkeqms  Error: None
+Driver JobID: 01jt6bmrthrtzgdwn7tad307nk  Error: None
+Lookout URL for this job is http://localhost:30000/?page=0&sort[id]=jobId&sort[desc]=true&ps=50&sb=01jt6bmrthrtzgdwn7tad307nk&active=false&refresh=true
+25/05/01 16:42:27 INFO ShutdownHookManager: Shutdown hook called
+25/05/01 16:42:27 INFO ShutdownHookManager: Deleting directory /tmp/spark-a5c32faa-d3e4-4d5d-82a6-cdb79bac7984
+```
+
+Note the Lookout URL. This will let us easily view results for the Driver and
+its Executors. If all goes well the jobs will succeed and produce a result!
+
+## Configuration Options
+
+`armada-spark` has a slew of configuration options that can be set with the
+`--conf` flag.
+
+Each option consists of a `String` unless otherwise noted.
+
+- `spark.armada.Executor.trackerPollingInterval` - Specifies
+    the interval between polls to check the state of Executors.
+- `spark.armada.Executor.trackerTimeout` - Specifies the time to
+    wait for the minimum number of Executors.
+- `spark.armada.lookouturl` - Sets the base URL to use for Armada's Lookout
+    service.
+- `spark.armada.health.checkTimeout` - Time to wait for Armada's health check
+    to succeed in seconds.
+- `spark.armada.clusterSelectors` - A comma-separated list of kubernetes
+    label selectors (in key=value format) to ensure the spark Driver and
+    its Executors are deployed to the same cluster.
+- `spark.armada.scheduling.nodeUniformityLabel` - A single kubernetes label to
+    apply to both Driver and Executors.
+- `spark.armada.DriverServiceNamePrefix` - Defines the Driver's service name
+    prefix within Armada. Must consist of lowercase a-z and '-' characters only.
+- `spark.armada.global.labels` - A comma-separated list of kubernetes labels
+    (in key=value format) to be added to all both the Driver and Executor pods.
+- `spark.armada.Driver.labels` - A comma-separated list of kubernetes labels
+    (in key=value format) to be added to the Driver pod.
+- `spark.armada.Executor.labels` A comma-separated list of kubernetes labels
+    (in key=value format) to be added to all Executor pods.
+
+# Building `armada-spark`
+
+## Scala Source
+
+`armada-spark` can be built from source using maven. See the [Building Armada Spark](https://github.com/armadaproject/armada-spark/blob/master/README.md#building-armada-spark)
+section of the README for more information.
+
+## Docker image
+
+`armada-spark` uses docker to produce container images that serve as spark
+Drivers and Executors. See the [Building Docker Images](https://github.com/ClifHouck/armada-spark/blob/master/README.md#building-docker-images)
```

Comment:
repo

Reply:
Addressed in 1dc03b4ad7656563ccce6a5fb9607fa2b8266382